From 98068e63339f9f18c817b840380c75a1ef327885 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Wed, 18 Sep 2013 12:56:33 +0800
Subject: [PATCH] ENGR00280148 CTS
 android.provider.cts.MediaStore_Images_MediaTest failed

Defer creation of the file until the app calls openFile() wont create its
parent directory, the CTS case will failed if no /sdcard/DCIM/Camera directory
ensureFileExists() will make sure its parent's directory be created

Revert "Don't create files when inserting entries"

This reverts commit 92013781ab1573eee3d5d75682f320fe3a6076f2.

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 src/com/android/providers/media/MediaProvider.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/com/android/providers/media/MediaProvider.java b/src/com/android/providers/media/MediaProvider.java
index 4528c56..9e50392 100755
--- a/src/com/android/providers/media/MediaProvider.java
+++ b/src/com/android/providers/media/MediaProvider.java
@@ -2766,7 +2766,9 @@ public class MediaProvider extends ContentProvider {
             values = initialValues;
         }
 
-        // we used to create the file here, but now defer this until openFile() is called
+        if (!ensureFileExists(file)) {
+            throw new IllegalStateException("Unable to create new file: " + file);
+        }
         return values;
     }
 
-- 
1.8.0

